[gingerbreaf] iP

gingerbreaf opened the pull request:
# Jerry 
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Jerry frees up your mind and helps you remember things. It's
- text-based
- easy to learn
- ~~Fast~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it
3. add your tasks.
4. let it manage your tasks for you 😃 

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Rename, Greet, Exit

Commit:
Echo + change file name

Commit:
Add, List

Commit:
Mark as Done + Task Class

Commit:
ToDos, Events, Deadlines + inheritance

Commit:
Automated Text UI Testing

Commit:
Error Handling

Commit:
Delete + ArrayList

Commit:
Implement hard disk saving

Commit:
Implement DateTime instead of using strings

Commit:
Merge branch 'branch-Level-8'

Commit:
Add more OOP, create Ui, TaskList, Command, Parser

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Place all files into a single package: Jerry

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge branch 'add-gradle-support' of github.com:gingerbreaf/ip

Commit:
Add JUnit test cases for Parser, Storage and TaskList

Commit:
Merge branch 'branch-A-JUnit'

Commit:
Add different command types, handle command exceptions, and other improvements

Commit:
Add JavaDocs comments

Commit:
Update to coding standards

Commit:
Implement find command

Commit:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

Commit:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Update config files and build,gradle

tengcharmaine commented on src/main/java/Jerry/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+
+    public boolean byIsNull() {
+        return by == null;
+    }
```
Can add JavaDocs for this method

tengcharmaine commented on src/main/java/Jerry/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+        return "D | " + (isDone? 1 : 0) + " | " + description + " | " + by;
+    }
+
+    public boolean byIsNull() {
```
A suggestion for the naming of this method is isByNull.

tengcharmaine commented on src/main/java/Jerry/Event.java:
```diff
@@ -0,0 +1,53 @@
+
+    public boolean dateTimeIsNull() {
+        return from == null || to == null;
+    }
```
Can add JavaDocs for this and also a naming suggestion, isDateTimeNull.

tengcharmaine commented on src/main/java/Jerry/Parser.java:
```diff
@@ -0,0 +1,88 @@
+package Jerry;
+
+import Jerry.command.*;
```
Should not use wildcard imports (ie list all the imports out).

tengcharmaine commented on src/main/java/Jerry/Storage.java:
```diff
@@ -0,0 +1,116 @@
+        return tasks;
+    }
+
+    public static Task parseTask(String line) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Task.java:
```diff
@@ -0,0 +1,82 @@
+     * Returns the description of this task.
+     *
+     * @return A string representing the task's description.
+     */
```
JavaDocs not required for getters

tengcharmaine commented on src/main/java/Jerry/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+        return this.tasks.remove(index);
+    }
+
+    public int size() {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+        return tasks.size();
+    }
+
+    public String printTask(int index) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+        return tasks.get(index).toString();
+    }
+
+    public void mark(int index) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+        tasks.get(index).markDone();
+    }
+
+    public void unmark(int index) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Hello! I'm Jerry.\nWhat can I do for you?");
+    }
+
+    public void showGoodbye() {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public void showError(String message) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Error: " + message);
+    }
+
+    public void showLoadingError() {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Error loading tasks from file.");
+    }
+
+    public void showMessage(String message) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        }
+    }
+
+    public void showList(TaskList list) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        }
+    }
+
+    public void showMark(TaskList list, int taskIndex) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Nice! I've marked this task as done:\n  " + tasks.get(taskIndex));
+    }
+
+    public void showUnmark(TaskList list, int taskIndex) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("OK, I've marked this task as not done yet:\n  " + tasks.get(taskIndex));
+    }
+
+    public void showDelete(TaskList list, Task removedTask) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Now you have " + (tasks.size()) + " tasks in the list.");
+    }
+
+    public void showAdded(Task task, TaskList tasks) {
```
Can add JavaDocs for this method.

tengcharmaine commented on src/main/java/Jerry/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+    private String commandDetails;
+
+    public AddDeadlineCommand(Ui ui, TaskList tasks, String commandDetails) {
```
Can add JavaDocs for this method (for the other methods in the other classes below also).

tengcharmaine commented on src/main/java/Jerry/command/Command.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public abstract void execute();
+//    public enum CommandType {
```
Remove these commented out lines if not needed.

tengcharmaine commented on src/main/java/Jerry/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,34 @@
+        this.commandDetails = commandDetails;
+    }
+
+    @Override
```
Can add JavaDocs for this method because it is quite different from the one in the super class (for the rest of your classes that extend from Command super class also)

tengcharmaine commented on src/main/java/Jerry/command/CommandFormatException.java:
```diff
@@ -0,0 +1,7 @@
+package Jerry.command;
+
+public class CommandFormatException extends Exception {
```
Can add JavaDocs for this class (and the other classes below).

tengcharmaine reviewed (commented):
Overall, I can understand what your code is doing. Just that some classes and methods are missing JavaDocs and code that is not needed (commented out) can be removed also.

donwong2308 commented on src/main/java/Jerry/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+     * @param keyword The keyword to search for in task descriptions.
+     * @return A list of tasks that contain the keyword.
+     */
+    public ArrayList<Task> findTasks(String keyword) {
+        return tasks.stream()
+                .filter(task -> task.getDescription().contains(keyword))
+                .collect(Collectors.toCollection(ArrayList::new));
```
I like the nice and clean implementation of the findTasks method.

donwong2308 commented on src/main/java/Jerry/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+ * Represents a deadline task in the chatbot application. A deadline task is a task that needs to be done before a specific date/time.
+ */
+public class Deadline extends Task {
+    private LocalDate by;
```
Perhaps this variable can be named as deadlineDate?

donwong2308 commented on src/main/java/Jerry/Event.java:
```diff
@@ -0,0 +1,53 @@
+    @Override
+    public String toFileFormat() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        return "E | " + (isDone ? 1 : 0) + " | " + description + " | " + from.format(formatter) + " | " + to.format(formatter);
```
Should your line length be longer than 120 chars?
```suggestion
        return "E | " + (isDone ? 1 : 0) + " | " + description + " | " 
                + from.format(formatter) + " | " + to.format(formatter);
```

donwong2308 commented on src/main/java/Jerry/Parser.java:
```diff
@@ -0,0 +1,88 @@
+                return new UnmarkCommand(ui, taskList, taskIndex);
+
+            case "delete":
+                taskIndex = Integer.parseInt(parts[1]) - 1;
+                return new DeleteCommand(ui, taskList, taskIndex);
+
+            case "todo":
+                if (parts.length < 2) {
+                    // Handle the case where the argument is missing
+                    return new AddTodoCommand(ui, taskList, "");
+                } else {
+                    return new AddTodoCommand(ui, taskList, parts[1]);
+                }
+
+            case "deadline":
+                if (parts.length < 2) {
+                    // Handle the case where the argument is missing
+                    return new AddDeadlineCommand(ui, taskList, "");
+                } else {
+                    return new AddDeadlineCommand(ui, taskList, parts[1]);
+                }
+
+            case "event":
+                if (parts.length < 2) {
+                    // Handle the case where the argument is missing
+                    return new AddEventCommand(ui, taskList, "");
+                } else {
+                    return new AddEventCommand(ui, taskList, parts[1]);
+                }
+
+            case "find":
+                if (parts.length < 2) {
+                    // Handle the case where the argument is missing
+                    return new FindCommand(ui, taskList, "");
+                } else {
+                    return new FindCommand(ui, taskList, parts[1]);
+                }
+            default:
+                return new InvalidCommand(ui);
+        }
```
I like the use the switch case here as it makes the code easy to understand and debug 👍

donwong2308 reviewed (commented):
Overall, looks good to me! Just some minor coding standard violations.

Commit:
Implement GUI functionality.

Commit:
Merge branch 'branch-Level-10'

Commit:
Introduce Assert statements to enforce assumptions

Assumptions about some operations such as adding or deleting need to be asserted to pinpoint out possible errors. Details about changes include assertions for checking whether tasklist for commands is null, and whether add and delete were successful.

Commit:
Clean up code to code quality standards
Change package name from Jerry to jerry, some variables to final, and remove commented code

gingerbreaf committed:
Merge pull request #2 from gingerbreaf/branch-A-Assertions

Introduce Assert statements to enforce assumptions

Commit:
Merge branch 'master' into branch-A-CodeQuality

gingerbreaf committed:
Merge pull request #3 from gingerbreaf/branch-A-CodeQuality

Clean up code to code quality standards

Commit:
Add view schedule functionality

Commit:
Merge branch 'branch-B-ViewSchedules'

Commit:
Update README.md

Commit:
Final changes

Commit:
Update build.gradle

Commit:
Update build.gradle for application main class
